modules/localgov_login_redirect/src/Form/LoginRedirectSettingsForm.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Drupal\localgov_login_redirect\Form;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Routing\RouteProvider;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+
+/**
+ * Login redirect settings for class.
+ */
+class LoginRedirectSettingsForm extends ConfigFormBase {
+
+  /**
+   * The route provider.
+   *
+   * @var \Drupal\Core\Routing\RouteProvider
+   */
+  protected RouteProvider $routeProvider;
+
+  /**
+   * Constructs a new UserRedirectSettingsForm.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The factory for configuration objects.
+   * @param \Drupal\Core\Routing\RouteProvider $route_provider
+   *   The route provider.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, RouteProvider $route_provider) {
+    parent::__construct($config_factory);
+    $this->routeProvider = $route_provider;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('router.route_provider')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'localgov_login_redirect_settings_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['localgov_login_redirect.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+
+    $form = parent::buildForm($form, $form_state);
+    $config = $this->config('localgov_login_redirect.settings');
+
+    $form['enabled'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Enable login redirect'),
+      '#description' => $this->t('Redirect users to the content overview page at login.'),
+      '#default_value' => $config->get('enabled'),
+    ];
+    $form['redirect_route'] = [
```

Comment:
If this is to solve the primary request (redirect to content list) it makes sense; but to generalize the chance is it'll need arguments (like user id) so tokens or so?

Reply:
Yep - good thought. Was keeping it simple. Not sure how much work it is to add token substitution to this, but can look into it if we're happy with the approach here.

Reply:
Yep - good thought. Was keeping it simple. Not sure how much work it is to add token substitution to this, but can look into it if we're happy with the approach here.

Reply:
I'm not sure quite what would be required to allow tokens in the route. Is this required for a first iteration of this or can it be put off for now?